CodingConventions.md

```diff
@@ -0,0 +1,145 @@
+# Overview
+
+We would like to be more focused on following the coding conventions popular in .NET community.
+Currently the coding style in the driver code is not very consistent, but the goal is to improve it
+by following coding convention rules.
+
+# Coding conventions
+
+## Writing tests
+
+### Arrange, Act, Assert pattern (3a)
+
+If possible split the test code into `arrange`, `act` and `assert` blocks.
+
+```csharp
+// arrange
+var config = new HttpClientConfig(
+    true,
+    "snowflake.com",
+    "123",
+    "testUser",
+    "proxyPassword",
+    "localhost", 
+    false,
+    false
+);
+
+// act
+var handler = (HttpClientHandler) HttpUtil.Instance.SetupCustomHttpHandler(config);
+
+// assert
+Assert.IsTrue(handler.UseProxy);
+Assert.IsNotNull(handler.Proxy);
+```
+
+### TestThatSomethingShouldHappen methods
+
+Use test names in Pascal notation (but without MS proposed underline characters between logical parts of the test name addressing 3a pattern).
+
+```csharp
+[Test]
+public void TestThatLoginWithInvalidPassowrdFails() {
+}
+
+[Test]
+public void TestCreatingHttpClientHandlerWithProxy() {
+}
+```
+
+## Writing code
+
+### Static variables
+
+#### Internal or private static member
+
+Use `s_` prefix + CamelCase, eg. `s_someVariable`.
```

Comment:
are there any tools that could be helpful to force convention in code in new files?

Reply:
Hard question. Even if we get a tool, currently the code is not following theses conventions so we will not be able to turn it on. The goal of this PR is only to collect coding convention we would like to follow.